### PR TITLE
feat(mongoose-id-validator): new definitions

### PR DESCRIPTION
**Pull request by kerolloz**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/mongoose-id-validator/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+     * entries when checking the database.
+     *
+     * Defaults to false
+     */
+    allowDuplicates?: boolean;
+}
+
+export function mongooseIdValidator(schema: Schema, options?: MongooseIdValidatorOptions): void;
+
+export default mongooseIdValidator;
```
Comment (amcasey) [inline types/mongoose-id-validator/index.d.ts:36]: Did you mean `exports = mongooseIdValidator`?  That appears to be how the underlying library exports it.

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/mongoose-id-validator/index.d.ts (hunk 1 above)
Comment (kerolloz) [reply types/mongoose-id-validator/index.d.ts:36]: The truth is I don't know :sweat_smile:. This is my first time to write Typings for a package. 
I thought it works when I tried it in the tests file. I can easily fix it, please let me know because I am not sure how to do it.

Comment (amcasey) [reply types/mongoose-id-validator/index.d.ts:36]: There are some notes here: https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require
TL;DR: Please use `exports = mongooseIdValidator;` in source and `import mongooseIdValidator = require('mongoose-id-validator');` in the tests.

Comment (kerolloz) [reply types/mongoose-id-validator/index.d.ts:36]: image
image
```
Argument of type 'typeof import("/home/codespace/workspace/DefinitelyTyped/types/mongoose-id-validator/index")' is not assignable to parameter of type '(schema: Schema<any>) => void'.
  Type 'typeof import("/home/codespace/workspace/DefinitelyTyped/types/mongoose-id-validator/index")' provides no match for the signature '(schema: Schema<any>): void'. ts(2345)
```

Am I doing something wrong? @amcasey

Comment (amcasey) [reply types/mongoose-id-validator/index.d.ts:36]: My bad, there's no 's' - it's just `export =`.  Sorry about that.

Comment (amcasey) [reply types/mongoose-id-validator/index.d.ts:36]: I should also explicitly mention that you'll want to drop the other `export`s, which will require the insertion of `declare` before `function`.

Comment (kerolloz) [reply types/mongoose-id-validator/index.d.ts]: Done. :heavy_check_mark:  
Thank you @amcasey.

**Review by amcasey — APPROVED** [review]
Thanks for the changes!

**Comment by kerolloz** [comment]
Ready to merge
